### PR TITLE
feat(payment): INT-6297 BNZ GooglePay

### PR DESCRIPTION
**Pull request by BC-PabloRivera**
## What?  [INT-6297](https://bigcommercecloud.atlassian.net/browse/INT-6297)
Googlepay implementation for Bank of New Zealand

## Why?
So that customers can pay with their google pay wallets

## Testing / Proof
...

@bigcommerce/checkout @bigcommerce/payments

### CODE REVIEW

File: packages/core/src/checkout-buttons/checkout-button-options.ts [hunk 1]
```diff
@@ -70,6 +70,12 @@ export interface BaseCheckoutButtonInitializeOptions extends CheckoutButtonOptio
      */
     googlepayadyenv3?: GooglePayButtonInitializeOptions;
 
+    /**
+     * The options that are required to facilitate CybersourceV2 GooglePay. They can be
```
Comment (vmparra) [inline packages/core/src/checkout-buttons/checkout-button-options.ts:74]: CybersourceV2? I understand this integration is based on CybersourceV2's and we might have been end it up copy & paste big pieces of code and docs. There are 50 matches for this wording, please make sure to rename (we applies).

**Comment by BC-PabloRivera** [comment]
♻️ @vmparra I changed all unnecessary cybs references

**Review by vmparra — APPROVED** [review]
LGTM 👍🏻

File: packages/core/src/payment/strategies/googlepay/googlepay-bnz-initializer.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                    parameters: {
+                        gateway: 'cybersource',
+                        gatewayMerchantId,
+                    },
+                },
+            }],
+            transactionInfo: {
+                currencyCode,
+                totalPriceStatus: 'FINAL',
+                totalPrice: round(outstandingBalance, 2).toFixed(2),
```
Comment (animesh1987) [inline packages/core/src/payment/strategies/googlepay/googlepay-bnz-initializer.ts:99]: Should we respect the currency settings here?

**Review by animesh1987 — APPROVED** [review]

**Pull request merged**